lib/simplecov.rb

```diff
@@ -46,15 +46,52 @@ class << self
     #
     def start(profile = nil, &block)
       require "coverage"
-      load_profile(profile) if profile
-      configure(&block) if block_given?
+      initial_setup(profile, &block)
       @result = nil
-      self.running = true
       self.pid = Process.pid
 
       start_coverage_measurement
     end
 
+    #
+    # Collate a series of SimpleCov result files into a single SimpleCov output.
+    # You can optionally specify configuration with a block:
+    #   SimpleCov.collate Dir["simplecov-resultset-*/.resultset.json"]
+    #    OR
+    #   SimpleCov.collate Dir["simplecov-resultset-*/.resultset.json"], 'rails' # using rails profile
+    #    OR
+    #   SimpleCov.collate Dir["simplecov-resultset-*/.resultset.json"] do
+    #     add_filter 'test'
+    #   end
+    #    OR
+    #   SimpleCov.collate Dir["simplecov-resultset-*/.resultset.json"], 'rails' do
+    #     add_filter 'test'
+    #   end
+    #
+    # Please check out the RDoc for SimpleCov::Configuration to find about
+    # available config options, or checkout the README for more in-depth
+    # information about coverage collation
+    #
+    def collate(result_filenames, profile = nil, &block)
+      if result_filenames.empty?
+        $stderr.printf("There's no reports to be merged\n")
```

Comment:
Maybe we'd rather wanna raise here? In that case I'd also probably convert it to a guard clause. What do you think?

Reply:
Addressed by https://github.com/colszowka/simplecov/pull/780/commits/2f01dff18ace47f393988eec5680249b995d4970.